### ISSUE TITLE
Acquiring further missions in chains before completing current missions

### ISSUE THREAD

**Issue by Zmedaris** [CleverRaven/Cataclysm-DDA#32145]
NOT a very concise description, but it's a bit of a complicated problem.

I just walked into my evac shelter base, spoke to one of my NPCs, and he gave me the 'intro conversation', as if we'd Just met, and then offered me a mission that was the next one in his chain. In this case, I received the mission to kill a zombie leader before I finished killing 100 zombies.

Out of curiosity, I spoke to the other two NPCs I have. They didn't do the 'intro conversation', but when I asked for a job, they both gave me the next mission in their chain as well: military blackbox before finding a flag, and the second mission in the 'religious' chain before I found the relic.

# Steps To Reproduce
Unfortunately, I have no idea how to get the NPCs to actually do this. It's only happened to me twice over the past two weeks or so. BUT, I will provide my current save where I experienced the above situation, as a case study that might help devs figure it out. 

# Expected behavior
I should not be able to acquire further missions in a chain before completing the current missions.

# Screenshot
![missionbug](https://user-images.githubusercontent.com/12032854/60712878-ce1c7800-9ecc-11e9-9d78-5b8c091932ad.png)

# Versions and configuration
- OS: Windows
    - OS Version: 10.0 1803
- Game Version: 0.D-5245-gaca7cde [64-bit]
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    Aftershock [aftershock],
    C.R.I.T Expansion Mod [crt_expansion],
    Magiclysm [magiclysm],
    Makeshift Items Mod [makeshift],
    Medieval and Historic Classes and Shields [Medieval_Stuff],
    More Survival Tools [More_Survival_Tools],
    Modular Turrets [modular_turrets],
    Salvaged Robots [Salvaged_Robots],
    Alternative Map Key [alt_map_key],
    Hydroponics [hydroponics],
    Mutant NPCs [mutant_npcs],
    Mythical Martial Arts [MMA],
    Folding Parts pack [deoxymod],
    Vehicle Additions Pack [blazemod],
    Tanks and Other Vehicles [Tanks],
    Disable NPC Needs [no_npc_food],
    SpeedyDex [speedydex],
    StatsThroughSkills [StatsThroughSkills],
    No Rail Stations [No_Rail_Stations]
]

# Additional context
I had this happen once before and didn't think much of it, but now I'm pretty sure it's a recurring problem.

It's also worth noting: the one that did his 'introduction event' conversation is the only one where I've actually completed a mission. I have killed the Jabberwock, and he presented this behavior soon after I got the second quest. (As in, the immediate next time I spoke with him.) I'm not sure if he did this because it's the type of mission where he joined me right after I finished the mission, or just because he's the only one I've actually completed a mission for, but it was 'different' than the other two whom I had to ASK for the missions, first.

# Save File
[Save.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/3362013/Save.zip)

**Comment by Zmedaris** [comment]
Small update: I had a thought, and decided to teleport to a school and grab a flag real quick.

Apparently I can't even properly complete the earlier missions while I have the later missions in my log.

**Comment by TychaiosIlithios** [comment]
Have some mess with npcs and missions atm - it involves randomly having to ask them to rejoin party after asking for a mission. On one hand no prob - request is 100%, on the other it looks sorta kinda buggy.

...aaand mentioned it in your thread 'cuz it's also about minor mess with npcs and missions.

**Comment by Zmedaris** [comment]
> Have some mess with npcs and missions atm - it involves randomly having to ask them to rejoin party after asking for a mission. On one hand no prob - request is 100%, on the other it looks sorta kinda buggy.
> 
> ...aaand mentioned it in your thread 'cuz it's also about minor mess with npcs and missions.

Yes, that's pretty much exactly what happened to me before that same NPC tried to push onto me a mission I should not have had access to yet.
